Build Fails due to Operator Overload Warnings

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

## Description

When running `make install`, the build fails when trying to build `DCSAM.cpp.o`. Note that there are no errors, and I got the build errors because my compiler was set to treat all warnings as errors. This was a fresh install of dependencies and a new build, and I don't think I put the flag `overloaded-virtual` anywhere.

This is happening because `DiscreteBoundaryFactor` and the base class `DecisionTreeFactor` overload the same operators `*` and `/`. This causes the base class version of the overload to be hidden unless explicitly brought into scope.

## Steps to reproduce

1. Navigate to `build` directory
2. Run `cmake ..`
3. Run `make install`

```
olive@olive-ubuntu-24:~/gtsam/build$ make install
[ 26%] Built target metis-gtsam
[ 26%] Built target cephes-gtsam
[ 26%] Built target CppUnitLite
[ 26%] Building CXX object gtsam/CMakeFiles/gtsam.dir/hybrid/DCSAM.cpp.o
In file included from /home/olive/gtsam/gtsam/discrete/DiscreteFactorGraph.h:22,
                 from /home/olive/gtsam/gtsam/hybrid/DCSAM.h:23,
                 from /home/olive/gtsam/gtsam/hybrid/DCSAM.cpp:20:
/home/olive/gtsam/gtsam/discrete/DecisionTreeFactor.h:174:24: error: ‘virtual gtsam::DecisionTreeFactor gtsam::DecisionTreeFactor::operator*(const gtsam::DecisionTreeFactor&) const’ was hidden [-Werror=overloaded-virtual=]
  174 |     DecisionTreeFactor operator*(const DecisionTreeFactor& f) const override {
      |                        ^~~~~~~~
In file included from /home/olive/gtsam/gtsam/hybrid/DCSAM.cpp:21:
/home/olive/gtsam/gtsam/hybrid/DiscreteBoundaryFactor.h:102:26: note:   by ‘gtsam::DiscreteBoundaryFactor::operator*’
  102 |   DiscreteBoundaryFactor operator*(const DiscreteBoundaryFactor& f) const;
      |                          ^~~~~~~~
/home/olive/gtsam/gtsam/discrete/DecisionTreeFactor.h:194:32: error: ‘virtual gtsam::DiscreteFactor::shared_ptr gtsam::DecisionTreeFactor::operator/(const gtsam::DiscreteFactor::shared_ptr&) const’ was hidden [-Werror=overloaded-virtual=]
  194 |     DiscreteFactor::shared_ptr operator/(
      |                                ^~~~~~~~
/home/olive/gtsam/gtsam/hybrid/DiscreteBoundaryFactor.h:113:26: note:   by ‘gtsam::DiscreteBoundaryFactor gtsam::DiscreteBoundaryFactor::operator/(const gtsam::DiscreteBoundaryFactor&) const’
  113 |   DiscreteBoundaryFactor operator/(const DiscreteBoundaryFactor& f) const;
      |                          ^~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [gtsam/CMakeFiles/gtsam.dir/build.make:1336: gtsam/CMakeFiles/gtsam.dir/hybrid/DCSAM.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3553: gtsam/CMakeFiles/gtsam.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

## Expected behavior
Build Succeeds.

## Environment
- Ubuntu 24.04.2 LTS
- GCC 13.3.0
- CMake 3.28.3

## Additional information

<!-- Add any other infor or context about the problem here. -->

--- a/gtsam/discrete/DiscreteConditional.h
+++ b/gtsam/discrete/DiscreteConditional.h
@@ -123,6 +123,8 @@ class GTSAM_EXPORT DiscreteConditional
                       const DecisionTreeFactor& marginal,
                       const Ordering& orderedKeys);
 
+  using DecisionTreeFactor::operator*;
+
   /**
    * @brief Combine two conditionals, yielding a new conditional with the union
    * of the frontal keys, ordered by gtsam::Key.

--- a/gtsam/hybrid/DiscreteBoundaryFactor.h
+++ b/gtsam/hybrid/DiscreteBoundaryFactor.h
@@ -95,11 +95,15 @@ class GTSAM_EXPORT DiscreteBoundaryFactor : public DecisionTreeFactor {
   /// @name Standard Interface
   /// @{
 
+  using DecisionTreeFactor::operator*;
+
   /// multiply with a scalar
   DiscreteFactor::shared_ptr operator*(double s) const override;
 
   /// multiply two factors
   DiscreteBoundaryFactor operator*(const DiscreteBoundaryFactor& f) const;
+
+  using DecisionTreeFactor::operator/;
 
   /**
    * @brief Divide by factor f (safely).

